content/2017.03.building-projects-as-a-way-to-learn.rst

```diff
@@ -0,0 +1,200 @@
+
+However, we can build software that, in one hand, is useful for the
+Mozilla mission and makes us reactive to fix Mozilla issues while
+building Firefox, and in the other hand enpower developers to build
+great software where users choose where they want to store their data.
+
+What matters rather than building software is the vision that this
+software carries.
+
+Writting lines of code is cheap and fast. You can build a prototype in
+two weeks and iterate until you are happy with the result.
+
+Then it might be use by some people and it might get some traction for
+few months or few years and it will be eventually forgotten in a git
+repository somewhere forever.
+
+Writting software to solve problems is not enough, to keep motivated
+on the long term.
+
+Our understanding is that you need a greater goal than just writting
+code.
+
+For us the answer to that question is: what can we learn while
+building software that can be useful for us and for our community to
+help us go faster and build more robust software later.
+
+For Mozilla the answer to that question is: How do I improve Firefox
+so that people want to use it as their main browser?
+
+Both answer are goals that we set to ourselves and we are using
+software development to reach them.
+
+Mozilla is calling that the True North, we are calling it a vision,
+some other people might call it an utopia.
+
+In both cases that vision should set the route of our ambition and give
+people a aim to all look in the same direction.
+
+
+How do you turn this vision into OKRs?
+======================================
+
+OKR means **Objectives and Key results**, it is a way to define
+tangible objectives and key results. You can then look at the results
+to tell if you were able to achieve your objectives.
+
+It is also a way to cut your vision in steps and to move your vision
+forward.
+
+OKR should be defined for quite short period of time to stay agile and
+be able to adapt to new environment constraints.
+
+At Mozilla we define quaterly OKRs.
+
+
+How do you set course on a vision?
+==================================
+
+To be able to define OKR from your vision, you need to be able to find
+the fastest way to get closer to your aim, however what we can learn
+from sailing is that the fastest way is rarely the shortest.
+
+At Mozilla, we have a vision that is written down in the
+`Mozilla Manifesto <https://www.mozilla.org/en-US/about/manifesto/>`_.
+
+Mitchell Baker does a great job making sure everyone at Mozilla stays
+aligned with that vision.
+
+However agreeing on the fastest way and turning the Mozilla vision
+into OKR is really challenging and even more when you have a thousand
+people opinion about it.
+
+That's the board job to set course every six months (or even less) on
+what we should be focusing on for the next six months.
+
+This often means changing our course and killing projects or at least
+putting them on the side if team are turning over to new priorities.
```

Comment:
Although this section asks "how do you do ...?", it ends with "The board does it for us", which is kind of unsatisfying.